applications/CompressiblePotentialFlowApplication/tests/cpp_tests/test_compressible_potential_element.cpp

```diff
@@ -0,0 +1,335 @@
+      rModelPart.GetProcessInfo()[HEAT_CAPACITY_RATIO] = 1.4;
+      rModelPart.GetProcessInfo()[SOUND_VELOCITY] = 340.0;
+
+      // Geometry creation
+      rModelPart.CreateNewNode(1, 0.0, 0.0, 0.0);
+      rModelPart.CreateNewNode(2, 1.0, 0.0, 0.0);
+      rModelPart.CreateNewNode(3, 1.0, 1.0, 0.0);
+      std::vector<ModelPart::IndexType> elemNodes{ 1, 2, 3 };
+      rModelPart.CreateNewElement("CompressiblePotentialFlowElement2D3N", 1, elemNodes, pElemProp);
+    }
+
+    /** Checks the IncompressiblePotentialFlowElement element.
+     * Checks the LHS and RHS computation.
+     */
+    KRATOS_TEST_CASE_IN_SUITE(CompressiblePotentialFlowElementRHS, CompressiblePotentialApplicationFastSuite)
+    {
+      Model this_model;
+      ModelPart& model_part = this_model.CreateModelPart("Main", 3);
+
+      GenerateCompressibleElement(model_part);
+      Element::Pointer pElement = model_part.pGetElement(1);
+
+      // Define the nodal values
+      Vector potential(3);
+      potential(0) = 1.0;
+      potential(1) = 2.0;
+      potential(2) = 3.0;
+
+      for (unsigned int i = 0; i < 3; i++){
+        pElement->GetGeometry()[i].FastGetSolutionStepValue(VELOCITY_POTENTIAL) = potential(i);
+      }
+      // Compute RHS and LHS
+      Vector RHS = ZeroVector(3);
+      Matrix LHS = ZeroMatrix(3, 3);
+
+      pElement->CalculateLocalSystem(LHS, RHS, model_part.GetProcessInfo());
+
+      std::vector<double> reference({0.615562, 0.0, -0.615562});
+
+      for (unsigned int i = 0; i < RHS.size(); i++) {
+        KRATOS_CHECK_NEAR(RHS(i), reference[i], 1e-5);
+      }
+    }
+
+    /** Checks the IncompressiblePotentialFlowElement element.
+     * Checks the LHS and RHS computation.
+     */
+    KRATOS_TEST_CASE_IN_SUITE(CompressiblePotentialFlowElementLHS, CompressiblePotentialApplicationFastSuite)
+    {
+      Model this_model;
+      ModelPart& model_part = this_model.CreateModelPart("Main", 3);
+
+      GenerateCompressibleElement(model_part);
+      Element::Pointer pElement = model_part.pGetElement(1);
+
+      // Define the nodal values
+      Vector potential(3);
+      potential(0) = 1.0;
+      potential(1) = 2.0;
+      potential(2) = 3.0;
+
+      for (unsigned int i = 0; i < 3; i++){
+        pElement->GetGeometry()[i].FastGetSolutionStepValue(VELOCITY_POTENTIAL) = potential(i);
+      }
+      // Compute RHS and LHS
+      Vector RHS = ZeroVector(3);
+      Matrix LHS = ZeroMatrix(3, 3);
+
+      pElement->CalculateLocalSystem(LHS, RHS, model_part.GetProcessInfo());
+
+      Matrix reference = ZeroMatrix(3,3);
+
+      reference(0,0) = 0.615556;
+      reference(0,1) = -0.615562;
+
+      reference(1,0) = -0.615562;
+      reference(1,1) = 1.23112;
+      reference(1,2) = -0.615562;
+
+      reference(2,1) = -0.615562;
+      reference(2,2) = 0.615556;
+
+      for (unsigned int i = 0; i < reference.size1(); i++) {
+        for (unsigned int j = 0; j < reference.size1(); j++) {
```

Comment:
```suggestion
        for (unsigned int j = 0; j < reference.size2(); j++) {
```
I think that this is wrong. Actually you can store the values by using a stack array of the type std::array<double, 9> and then retrieve the values as `reference(i*3 + j)`